parquet/src/schema/types.rs

```
@@ -556,7 +556,8 @@ impl<'a> PrimitiveTypeBuilder<'a> {
                 }
             }
             PhysicalType::FIXED_LEN_BYTE_ARRAY => {
-                let max_precision = (2f64.powi(8 * self.length - 1) - 1f64).log10().floor() as i32;
+                let length = self.length.checked_mul(8).unwrap_or(i32::MAX);
```

should the overflow error instead of falling through to max precision?

---

I agree it would be better to return an error:

I double checked that using i32::MAX results in 

```
Max precision: 2147483647
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8b510b23f773540e2b348e80f0691b41

---

sg, changed to returning error, thanks!